### DISCUSSION TITLE
Hubs Cloud: "join crashed"

### DISCUSSION THREAD

**Discussion by DougReeder** [mozilla/hubs#3352] [Hubs Cloud Questions] [answered]
We're running an instance of Hubs Cloud, which works normally for others on our dev team. I can run `npm run login` and it reports 

```
Login successful.
Credentials written to .ret.credentials. Run npm run logout to remove credentials.
```
(.ret.credentials does, indeed appear to contain credentials.)

When I run `npm start` it reports "Compiled successfully.", but when I actually try to connect in Firefox, Chrome or Reality, it stalls on the "Loading Please Wait..." screen, and in the console there are repeated messages 
`Object { reason: "join crashed" }`
That error is thrown from line 305 in phoenix.js:
```
  matchReceive({status, response, ref}){
    this.recHooks.filter( h => h.status === status )
                 .forEach( h => h.callback(response) )
  }
```
...but I can't find "join crashed" anywhere in the front-end repo.

```
Object { reason: "join crashed" }
    overrideMethod react_devtools_backend.js:2430
    js hub.js:1510
    matchReceive phoenix.js:305
    matchReceive phoenix.js:305
    startTimeout phoenix.js:334
    trigger phoenix.js:646
    Channel phoenix.js:417
    trigger phoenix.js:646
    onConnMessage phoenix.js:1152
    decode phoenix.js:693
    onConnMessage phoenix.js:1143
    onmessage phoenix.js:878
    (Async: EventHandlerNonNull)
    connect phoenix.js:878
    _connectToReticulum phoenix-utils.js:125
    asyncGeneratorStep hub-06ca5dc166958bb75730.js:61326
    _next hub-06ca5dc166958bb75730.js:61328
    (Async: promise callback)
    asyncGeneratorStep hub-06ca5dc166958bb75730.js:61326
    _next hub-06ca5dc166958bb75730.js:61328
    _asyncToGenerator hub-06ca5dc166958bb75730.js:61328
    _asyncToGenerator hub-06ca5dc166958bb75730.js:61328
    _connectToReticulum hub-06ca5dc166958bb75730.js:61506
    connectToReticulum phoenix-utils.js:95
    js hub.js:1122
    asyncGeneratorStep hub-06ca5dc166958bb75730.js:20836
    _next hub-06ca5dc166958bb75730.js:20838
    (Async: promise callback)
    asyncGeneratorStep hub-06ca5dc166958bb75730.js:20836
    _next hub-06ca5dc166958bb75730.js:20838
    (Async: promise callback)
    asyncGeneratorStep hub-06ca5dc166958bb75730.js:20836
    _next hub-06ca5dc166958bb75730.js:20838
    (Async: promise callback)
    asyncGeneratorStep hub-06ca5dc166958bb75730.js:20836
    _next hub-06ca5dc166958bb75730.js:20838
    _asyncToGenerator hub-06ca5dc166958bb75730.js:20838
    _asyncToGenerator hub-06ca5dc166958bb75730.js:20838
    (Async: EventListener.handleEvent)
    js hub.js:707
    Webpack 6
```

What should we be looking at?

**Comment by DougReeder** [comment]
I was using a URL for a room on a different server.  :-(  Using just https://localhost:8080/ I can connect just fine.

So, what generates that error message? I'd be happy to write a PR with a more helpful error message.

**Comment by robinkwilson** [comment, marked as answer]
Are you basing your hubs-cloud custom client on the **hubs-cloud** branch (in the hubs repo)?

The **master** branch in the hubs repo is what is live on hubs.mozilla.com and is ahead of the front end code running on Hubs Cloud Machines. So we ask people hacking on custom clients to use the hubs-cloud branch.

I recently updated the guide to walk you through setting this up: https://hubs.mozilla.com/docs/hubs-cloud-custom-clients.html

Also the "join crashed" error likely came from the backend in the https://github.com/mozilla/reticulum repo. That's likely why you can't find it in the front end code.

**Reply by DougReeder** [reply]
The “join crashed” message is in the Phoenix source code, which makes it understandable that the error message is so vague. 

When I have a bit more time I’ll see about tapping the error and converting the message to something more useful.
